internal/runtime/catalog/test/catalog_test.go

```diff
@@ -0,0 +1,78 @@
+/*
+Copyright 2022 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+// Package test contains catalog tests
+// Note: They have to be outside the catalog package to be realistic. Otherwise using
+// test types with different versions would result in a cyclic dependency and thus
+// wouldn't be possible.
+package test
+
+import (
+	"testing"
+
+	. "github.com/onsi/gomega"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+
+	"sigs.k8s.io/cluster-api/internal/runtime/catalog"
+	"sigs.k8s.io/cluster-api/internal/runtime/catalog/test/v1alpha1"
+	"sigs.k8s.io/cluster-api/internal/runtime/catalog/test/v1alpha2"
+)
+
+var c = catalog.New()
+
+func init() {
+	_ = v1alpha1.AddToCatalog(c)
+	_ = v1alpha2.AddToCatalog(c)
+}
+
+func TestCatalog(t *testing.T) {
```

Comment:
Not for the POC but we definitely need more tests (not only the happy path) + some comment explaining what we are testing

Reply:
Absolutely!